### PR TITLE
refactor order api

### PR DESCRIPTION
**Pull request by leonthaibao**
Fix single-quote issues,
Refactor order APIs to the new style.
Update mustache template

### CODE REVIEW

File: src/zalopay/models/agreementBindResponse.ts [hunk 1]
```diff
@@ -8,61 +8,61 @@
...
+  */
+  "return_code"?: number;
 
-        'return_message'?: string;
+  "return_message"?: string;
 
-        /**
-        * 
-        */
-        'sub_return_code'?: number;
```
[the remark this answers is not in the document]
Comment (leonthaibao) [reply src/zalopay/models/agreementBindResponse.ts:21]: I have just fixed this, please re-review

File: .eslintrc.js [hunk 2]
```diff
@@ -32,7 +32,8 @@ module.exports = {
           "{}": false
         }
       }
-    ]
+    ],
+    "@typescript-eslint/no-namespace": "off"
```
Comment (longxice) [inline .eslintrc.js:36]: Should we allow to use namespace with the Typescript?
https://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html

File: src/zalopay/services/order.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    const data = [];
+    data.push(request.app_id);
+    data.push(request.app_trans_id);
+    data.push(request.app_user);
+    data.push(request.amount);
+    data.push(request.app_time);
+    data.push(request.embed_data);
+    data.push(request.item);
+    return data.join(HmacUtils.DATA_SEPARATOR);
+  }
```
[the remark this answers is not in the document]
Comment (anhldbk) [reply src/zalopay/services/order.ts:55]: @leonthaibao can we refactor the code as:


```js
// from
const data = [];
data.push(request.app_id);
data.push(request.app_trans_id);
data.push(request.app_user);
data.push(request.amount);
data.push(request.app_time);
data.push(request.embed_data);
data.push(request.item);

// to
const data = [ request.app_id, request.app_trans_id....];
```

[the remark this answers is not in the document]
Comment (qtdan) [reply src/zalopay/services/order.ts:55]: @leonthaibao , please create an issue then we can track this.
@longxice I think you need to find a solution for this function

```
private getDataToSign(request:
        AgreementBindRequest
        | AgreementUnbindRequest
        | AgreementPayRequest
        | AgreementQueryUserRequest
        | AgreementBalanceRequest
        | AgreementQueryRequest): string {
        const data = [];
        if (request instanceof AgreementBindRequest) {
```

**Comment by noradomi** [comment]
> Fix single-quote issues, Refactor order APIs to the new style. Update mustache template

@leonthaibao Hình như Github có cho custome description template cho PR, anh xem add vô thử nha. 

https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

File: src/zalopay/services/order.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (noradomi) [reply src/zalopay/services/order.ts:55]: @leonthaibao trong models.ts có func findCorrectType, anh check xem có dùng được không

File: src/zalopay/__tests__/__mocks__/order/responses.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+export const createOrderSuccess = {
+  "return_code": 1,
+  "return_message": "Giao dịch thành công",
```
Comment (qtdan) [inline src/zalopay/__tests__/__mocks__/order/responses.ts:3]: english please

File: src/zalopay/__tests__/__mocks__/refund/responses.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+export const refundCreateSuccess = {
+  "return_code": 1,
+  "return_message": "Giao dịch thành công",
```
Comment (qtdan) [inline src/zalopay/__tests__/__mocks__/refund/responses.ts:3]: english please

File: src/zalopay/__tests__/__mocks__/refund/responses.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  "sub_return_code": 1,
+  "sub_return_message": "Giao dịch thành công",
+  "refund_id": 123456789,
+};
+
+export const refundQuerySuccess = {
```
Comment (qtdan) [inline src/zalopay/__tests__/__mocks__/refund/responses.ts:9]: this response looks weird to me, why we don't have any data on it?

File: src/zalopay/services/order.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (longxice) [reply src/zalopay/services/order.ts:55]: @leonthaibao , @noradomi , @qtdan dan
Đây là private func, chúng ta có tách biệt sign cho mỗi loại request khác nhau.

Nghĩa là, với mỗi request có hàm sign khác nhau. 
For example, 
`signAgreementBindRequest()`

File: src/zalopay/zaloPayClient.ts [hunk 7]
```diff
@@ -1,20 +1,17 @@
...
-  orderProvider: OrderProvider;
   securityProvider: SecurityProvider;
 
   constructor(config: Config) {
     this.config = config;
     this.httpClient = axios.create({
       baseURL: config.env == "sandbox" ? "https://sb-openapi.zalopay.vn" : "https://openapi.zalopay.vn",
     });
-    this.orderProvider = new OrderProvider(this);
     this.securityProvider = new SecurityProvider(this);
```
Comment (qtdan) [inline src/zalopay/zaloPayClient.ts:15]: I think `securityProvider` is no longer needed, can we remove?

File: src/zalopay/services/order.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (leonthaibao) [reply src/zalopay/services/order.ts:55]: > @leonthaibao trong models.ts có func findCorrectType, anh check xem có dùng được không

A có check logic func này, thì nó không sp được case của mình nha

File: src/zalopay/__tests__/__mocks__/refund/responses.ts (hunk 6 above)
Comment (leonthaibao) [reply src/zalopay/__tests__/__mocks__/refund/responses.ts:9]: This is the real data response when I tested on SB env

File: src/zalopay/__tests__/__mocks__/refund/responses.ts (hunk 5 above)
Comment (leonthaibao) [reply src/zalopay/__tests__/__mocks__/refund/responses.ts:3]: Okie I'll fix this

File: src/zalopay/zaloPayClient.ts (hunk 7 above)
Comment (leonthaibao) [reply src/zalopay/zaloPayClient.ts:15]: It's used to create a mac string to verify the callback data. I will refactor it later

File: .eslintrc.js (hunk 2 above)
Comment (leonthaibao) [reply .eslintrc.js]: I'm not sure about this, it's just used in the generated codes, not in our codes. so I think we can ignore this

File: src/zalopay/__tests__/__mocks__/refund/responses.ts (hunk 6 above)
Comment (noradomi) [reply src/zalopay/__tests__/__mocks__/refund/responses.ts:9]: @leonthaibao I thinks we can copy example response from our docs: https://docs.zalopay.vn/downloads/api/ZaloPay-APIs-Integration-Document.pdf.  
It'll be more consistent.

**Review by qtdan — APPROVED** [review]
LGTM

**Pull request merged**